htmd/builder/amber.py

```diff
@@ -372,10 +388,12 @@ def build(mol, ff=None, topo=None, param=None, prefix='structure', outdir='./bui
     if not ionize and len(disulfide) != 0:
         f.write('# Adding disulfide bonds\n')
         for d in disulfide:
+            atoms1 = d[0].selectAtoms(mol, ignore='resname')
+            atoms2 = d[1].selectAtoms(mol, ignore='resname')
             # Convert to stupid amber residue numbering
             uqseqid = sequenceID((mol.resid, mol.insertion, mol.segid)) + mol.resid[0]
-            uqres1 = int(np.unique(uqseqid[(mol.segid == d.segid1) & (mol.resid == d.resid1)]))
-            uqres2 = int(np.unique(uqseqid[(mol.segid == d.segid2) & (mol.resid == d.resid2)]))
+            uqres1 = int(np.unique(uqseqid[atoms1]))
+            uqres2 = int(np.unique(uqseqid[atoms2]))
             f.write('bond mol.{}.SG mol.{}.SG\n'.format(uqres1, uqres2))
```

Comment:
Unrelated with this PR, but we're passing the responsibility to tleap to error if the user provides wrong residues (residues that don't even have SG atoms)

Reply:
Hm yes I could add the check in this PR actually. It's marginal effort and nicer to catch it before shit hits the fan.

Reply:
I reconsidered. I think the builder could add the SG atom if it's missing. This would limit it a bit more. For example if you mutate the residue so it misses the sidechain atoms. I will leave it as it.